include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -550,6 +550,8 @@ class permutation_iterator
     typedef ::std::random_access_iterator_tag iterator_category;
     typedef ::std::true_type is_permutation;
 
+    permutation_iterator() = default;
+
     permutation_iterator(const SourceIterator& input1, const IndexMap& input2, ::std::size_t index = 0)
```

Comment:
Seems that this interface is not covered by test. Methods base() and map() are not covered either.

Reply:
The constructor is called by `make_permutation_iterator`, `permutation_iterator::operator+` and `permutation_iterator::operator-`. Do we require an explicit test of the constructor as well?

Reply:
@capatober, could you please express your opinion?

Reply:
I think we need to test these interfaces despite the fact that it's used in `make_permutation_iterator` because this is implementation details. As developers we know how it's implemented. However, what if we change implementation of `make_permutation_iterator` later? This interface will not be covered.
So, we need an explicit test since it's public interface and users can call it. We need to make sure that this interface is callable by users.

Reply:
Done.